src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -955,6 +957,15 @@ public static void populateBlackBox(Design design, String hierarchicalCellName,
                 }
             }
             design.addSiteInst(si);
+            // Update GND/VCC site routing to point to destination design's GND/VCC nets
+            for (String siteWire : si.getSiteWiresFromNet(vcc)) {
+                BELPin pin = si.getSiteWirePins(siteWire)[0];
+                si.routeIntraSiteNet(vcc, pin, pin);
+            }
```

Comment:
Hang on, isn't this redundant? You're asking for all sitewires from the `vcc` net, and then you're re-routing those sitewires onto the same `vcc` net?

Reply:
The problem is we are merging two different designs together.  They each have an instance of the GND and VCC nets.  If I don't update the reference pointers to the destination design's GND or VCC net, this method will fail because it checks references rather then `.equals()`:
https://github.com/Xilinx/RapidWright/blob/f6df54954f3280003360eb9af69d56bb01eb713f/src/com/xilinx/rapidwright/design/DesignTools.java#L3494

Reply:
`SiteInst.getSiteWiresFromNet()` uses `Net.equals()` under the hood so it relies on the name of the `Net` than the actual reference.